src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs

```diff
@@ -92,15 +94,28 @@ private void OnItemsChanged(IObservableVector<object> sender, IVectorChangedEven
 			UpdateSuggestionList();
 		}
 
+		internal override IEnumerable GetItems()
+		{
+			if (_ItemsSource?.Any() ?? false)
+			{
+				return _ItemsSource;
+			}
+			if (Items?.Any() ?? false)
+			{
+				return Items;
+			}
+			return Enumerable.Empty<object>();
+		}
+
 		protected override void OnItemsSourceChanged(DependencyPropertyChangedEventArgs e)
 		{
 			// Calling this method before base.OnItemsSourceChanged() ensures that, in the case of an ObservableCollection, the list
 			// subscribes to CollectionChanged before AutoSuggestBox does. This is important for Android because the list needs to
 			// notify RecyclerView of collection changes before UpdateSuggestionList() measures it, otherwise we get errors like
 			// "Inconsistency detected. Invalid view holder adapter position"
-			UpdateSuggestionList();
-
 			base.OnItemsSourceChanged(e);
+			_ItemsSource = (UnwrapItemsSource() as IEnumerable)?.ToObjectArray().AsEnumerable();
+			UpdateSuggestionList();
```

Comment:
This is somewhat concerning. The comment above mentions it was intentional that it's called before base.

Reply:
This is unnecessary.